src/components/Button/Button.styles.ts

```diff
@@ -0,0 +1,127 @@
+import { Button as MuiButton, CircularProgress } from '@mui/material';
+import { styled } from '@mui/material/styles';
+
+import {
+  blueScale,
+  redScale,
+  greenScale,
+  greyScale,
+} from '../../style-library/types/theme.helpers';
+
+import { SIZE_CONFIGS, SHAPE_CONFIGS } from './Button.constants';
+
+import type { StyledButtonProps } from './Button.types';
+
+// Map variant → palette
+const VARIANT_COLORS = {
+  primary: {
+    main: blueScale[600],
+    light: blueScale[50],
+    hover: blueScale[700],
+    active: blueScale[800],
+  },
+  secondary: {
+    main: greyScale[600],
+    light: greyScale[50],
+    hover: greyScale[700],
+    active: greyScale[800],
+  },
+  success: {
+    main: greenScale[600],
+    light: greenScale[50],
+    hover: greenScale[700],
+    active: greenScale[800],
+  },
+  error: { main: redScale[600], light: redScale[50], hover: redScale[700], active: redScale[800] },
+} as const;
+
+export const StyledButton = styled(MuiButton, {
+  shouldForwardProp: (prop) => !prop.toString().startsWith('$'),
+})<StyledButtonProps>(({ theme, $variant, $size, $shape, $loading, $fullWidth }) => {
+  const colors = VARIANT_COLORS[$variant];
+  const size = SIZE_CONFIGS[$size];
+  const shape = SHAPE_CONFIGS[$shape];
+
+  return {
+    height: size.height,
+    minWidth: $fullWidth ? '100%' : size.minWidth,
+    padding: size.padding,
+    fontSize: size.fontSize,
+    fontWeight: 500,
+    textTransform: 'none',
+    borderRadius: shape.borderRadius,
+    border: 'none',
+    position: 'relative',
+    overflow: 'hidden',
+
+    // contained
+    backgroundColor: colors.main,
+    color: '#fff',
```

Comment:
color is hard-coded to '#fff'. Hardcoded color literals should be avoided; use theme palette tokens to ensure consistent contrast and theming (e.g., theme.palette.common.white or theme.palette.getContrastText(background)).

**Sugerencia:**
```typescript
Use the theme contrast helper:
color: theme.palette.getContrastText(colors.main) || theme.palette.common.white,
```